### ISSUE TITLE
2.4.11: Gradient Focus Indicators

### ISSUE THREAD

**Issue by benja11y** [w3c/wcag#1861]
The following comment is made in the [Understanding Document](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible-enhanced):

> A keyboard focus indicator which has a pattern or gradient may have parts that do not meet the 3:1 contrast ratio for the change of contrast, as long as an area equal to the minimum does meet the contrast ratio.

With the current color contrast tooling available, testing multiple pixels of a gradational focus indicator to determine the area of the part which meets the required contrast ratio will be non-trivial, especially when non-standard shapes shapes are used.

The understanding document describes the process as straightforward, but we feel this would only be the case when the content under test is a clear pass. It would be significantly more laborious if the result is marginal. The precise nature of the test procedure may result in errors and therefore impact repeatability.

Editorial note: there are no spaces between the figure numbers and their captions.

This issue has been raised on behalf of [TPGi](https://www.tpgi.com), based on an internal consultation process. [An article containing all of our comments](https://www.tpgi.com/wcag-22-working-draft-feedback/) can be found on our blog, along with an introduction and acknowledgments.

**Comment by alastc** [comment]
> The understanding document describes the process as straightforward, but we feel this would only be the case when the content under test is a clear pass. It would be significantly more laborious if the result is marginal. 

Indeed, but what would the alternative be, just fail gradients? 

In our audits a gradient indicator is very, very rare. It hasn't come up in the last year (since we've been including this in testing, marked as 'future requirement'). Well, we've had a few that included a gradient, but they had a core indicator that passed, plus a superfluous gradient around it.

Given the rarity, is this a problem?

**Comment by alastc** [comment]
Proposed response:

<hr>

It may be difficult to test some focus-indicators which use a gradient. However, the group feels it is better to have a metric to test indicators than simply passing (or failing) gradient indicators.

In general, an indicator with a gradient can be tested, and if it contains a core of solid colour-change it could fulfil the criterion.
